# HG changeset patch
# Parent 694bd2b3a75093e4cf4c60876cf8804d2243edcc

diff -r 694bd2b3a750 hgblog/generate_html.py
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/hgblog/generate_html.py	Sat Apr 24 21:31:50 2010 -0400
@@ -0,0 +1,7 @@
+from hgblog.refresh import refresh
+
+def htmlize_articles(ui, repo, **kwargs):
+    """Calls on Sphinx to turn our .rst files into pretty HTML"""
+
+    refresh(repo)
+
