# HG changeset patch
# Parent 14ea69c060081944edea66ca8e111172d806fa63

diff --git a/.hgignore b/.hgignore
--- a/.hgignore
+++ b/.hgignore
@@ -14,3 +14,4 @@
 ^env/
 \.DS_Store$
 ~$
+.*\.swp
diff --git a/README b/README
--- a/README
+++ b/README
@@ -9,8 +9,8 @@
 
 Use ``setup.py``::
 
-   python setup.py build
-   sudo python setup.py install
+    python setup.py build
+    sudo python setup.py install
 
 
 Reading the docs
@@ -18,15 +18,16 @@
 
 After installing::
 
-   cd doc
-   sphinx-build . _build/html
+    cd doc
+    sphinx-build . _build/html
 
 Then, direct your browser to ``_build/html/index.html``.
 
-Or read them online at <http://sphinx.pocoo.org/>.
+Or read them online at <http://sphinx.pocoo.org>.
 
 
 Contributing
 ============
 
 Send wishes, comments, patches, etc. to sphinx-dev@googlegroups.com.
+
diff --git a/README.rst b/README.rst
new file mode 120000
--- /dev/null
+++ b/README.rst
@@ -0,0 +1,1 @@
+README
\ No newline at end of file
diff --git a/hgblog/__init__.py b/hgblog/__init__.py
new file mode 100644
--- /dev/null
+++ b/hgblog/__init__.py
@@ -0,0 +1,1 @@
+__version__ = '0.1a'
diff --git a/setup.py b/setup.py
--- a/setup.py
+++ b/setup.py
@@ -7,6 +7,7 @@
 from setuptools import setup, find_packages
 from distutils import log
 
+import hgblog
 import sphinx
 
 long_desc = '''
@@ -156,14 +157,14 @@
 
 
 setup(
-    name='Sphinx',
-    version=sphinx.__version__,
-    url='http://sphinx.pocoo.org/',
-    download_url='http://pypi.python.org/pypi/Sphinx',
+    name='hgblog',
+    version=hgblog.__version__,
+    url='http://bitbucket.org/codekoala/hgblog/',
+    download_url='http://pypi.python.org/pypi/hgblog',
     license='BSD',
-    author='Georg Brandl',
-    author_email='georg@python.org',
-    description='Python documentation generator',
+    author='Josh VanderLinden',
+    author_email='codekoala@gmail.com',
+    description='Sphinx-based blogging engine',
     long_description=long_desc,
     zip_safe=False,
     classifiers=[
@@ -182,12 +183,12 @@
     include_package_data=True,
     entry_points={
         'console_scripts': [
-            'sphinx-build = sphinx:main',
-            'sphinx-quickstart = sphinx.quickstart:main',
-            'sphinx-autogen = sphinx.ext.autosummary.generate:main',
+            'hgblog-build = sphinx:main',
+            'hgblog-quickstart = sphinx.quickstart:main',
+            'hgblog-autogen = sphinx.ext.autosummary.generate:main',
         ],
         'distutils.commands': [
-            'build_sphinx = sphinx.setup_command:BuildDoc',
+            'build_hgblog = sphinx.setup_command:BuildDoc',
         ],
     },
     install_requires=requires,
